bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/firmware/ProgressCallbackTest.groovy

```diff
@@ -0,0 +1,79 @@
+/**
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.thing.firmware
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.thing.ThingTypeUID
+import org.eclipse.smarthome.core.thing.ThingUID
+import org.eclipse.smarthome.core.thing.binding.firmware.FirmwareUID
+import org.eclipse.smarthome.core.thing.binding.firmware.ProgressCallback;
+import org.eclipse.smarthome.core.thing.binding.firmware.ProgressStep;
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * Testing the {@link ProgressCallback}.
+ *
+ * @author Christoph Knauf - Initial contribution
+ */
+public final class ProgressCallbackTest {
+
+    ProgressCallback sut
+    boolean eventPosted = false 
+    
+    @Before
+    void setUp(){
```

Comment:
I miss a test for a pending firmware update after a step has already been executed... furthermore we should test that a update can be resumed after it was pending